### PR TITLE
Separate database package mocks to dbmocks package

### PR DESCRIPTION
**Pull request by pjlast**
"Simpler" attempt at simplifying our `database` package. This moves all `database` mocks to its own `dbmocks` package, and updates all the tests to use the `dbmocks` package instead.

Functionally this is a non-change, and a first step at un-polluting our `database` package space.

Some of the tests in `database` were using the mocks (a package mocking itself to test itself is a bit sus ngl) and I edited those tests to either be a `database_test` package, or to use a real database.

## Test plan

Non-functional change, only test code affected. No tests functionally changed, all should still pass

### CODE REVIEW

File: internal/database/oauthtoken/oauth_token_helper.go [hunk 1]
```diff
@@ -1,10 +1,11 @@
-package database
+package oauthtoken
```
Comment (pjlast) [inline internal/database/oauthtoken/oauth_token_helper.go:1]: I moved these functions to their own package. They aren't database functions, they simply use the database.

File: internal/database/executor_secrets_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package database
+package database_test
```
Comment (pjlast) [inline internal/database/executor_secrets_test.go:1]: The executors secrets tests use a whole lot of mocks, so I moved them out to a `_test` test file

File: internal/database/executor_secrets_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package database
```
Comment (pjlast) [inline internal/database/executor_secrets_internal_test.go:1]: The executors secrets test that use an actual database and use internal functions I kept in the normal test package space.

File: internal/database/executor_secrets.go [hunk 4]
```diff
@@ -517,7 +517,7 @@ func scanExecutorSecret(secret *ExecutorSecret, key encryption.Key, s interface
 	return nil
 }
 
-func ensureActorHasNamespaceWriteAccess(ctx context.Context, db DB, secret *ExecutorSecret) error {
+func EnsureActorHasNamespaceWriteAccess(ctx context.Context, db DB, secret *ExecutorSecret) error {
```
Comment (pjlast) [inline internal/database/executor_secrets.go:520]: This is the only function I renamed so that it's public and accessible by the `_test` tests

**Review by pjlast — COMMENTED** [review]
Here are the changes that I made to solve the circular import issues. The rest of the changes are all pure `database -> dbmocks` renames

**Review by kopancek — APPROVED** [review]
LGTM :shipit: 🤷

**Review by sashaostrikov — APPROVED** [review]
Nice job! :shipit:

File: internal/database/oauthtoken/oauth_token_helper.go (hunk 1 above)
Comment (eseliger) [reply internal/database/oauthtoken/oauth_token_helper.go:1]: could this live in `internal/oauthtoken` too? this current path makes me believe it's a subfunction of the database, which, as you said here, is not true.

**Review by eseliger — APPROVED** [review]
from the comments you left, LGTM 👍 

Didn't look at every single of the 290 files though 😬

File: internal/database/oauthtoken/oauth_token_helper.go (hunk 1 above)
Comment (pjlast) [reply internal/database/oauthtoken/oauth_token_helper.go:1]: Yeah it can certainly be moved. I'll do it in a sec

**Review by efritz — APPROVED** [review]

**Review by camdencheek — APPROVED** [review]
Let's goooooooo

**Pull request merged**